examples/Example/src/new_api/bottom_sheet/index.tsx

```diff
@@ -0,0 +1,165 @@
+
+  const saveScrollOffset = (e: NativeSyntheticEvent<NativeScrollEvent>) => {
+    scrollOffset.value = e.nativeEvent.contentOffset.y;
+  };
+
+  const onPanHandlerUpdate = (e: PanGestureHandlerEventPayload) => {
+    translationY.value = e.translationY - scrollOffset.value;
+  };
+
+  const onHeaderHandlerUpdate = (e: PanGestureHandlerEventPayload) => {
+    translationY.value = e.translationY;
+  };
+
+  const onHeaderHandlerEnd = (e: PanGestureHandlerEventPayload) => {
+    onHandlerEnd(e);
+  };
+
+  const onPanHandlerEnd = (e: PanGestureHandlerEventPayload) => {
+    onHandlerEnd(e);
+  };
+
+  const onHandlerEnd = ({ velocityY }: PanGestureHandlerEventPayload) => {
+    const dragToss = 0.05;
+    const endOffsetY =
+      bottomSheetTranslateY.value + translationY.value + velocityY * dragToss;
+
+    // calculate nearest snap point
+    let destSnapPoint = FULLY_OPEN_SNAP_POINT;
+
+    if (
+      snapPoint === FULLY_OPEN_SNAP_POINT &&
+      endOffsetY < FULLY_OPEN_SNAP_POINT
+    ) {
+      return;
+    }
+
+    for (const snapPoint of SNAP_POINTS_FROM_TOP) {
+      const distFromSnap = Math.abs(snapPoint - endOffsetY);
+      if (distFromSnap < Math.abs(destSnapPoint - endOffsetY)) {
+        destSnapPoint = snapPoint;
+      }
+    }
+
+    // update current translation to be able to animate withSpring to snapPoint
+    bottomSheetTranslateY.value =
+      bottomSheetTranslateY.value + translationY.value;
+    translationY.value = 0;
+
+    bottomSheetTranslateY.value = withSpring(destSnapPoint, {
+      mass: 0.5,
+    });
+
+    setSnapPoint(destSnapPoint);
+  };
+
+  const blockScrollUntilAtTheTop = Gesture.Tap()
+    .maxDeltaY(snapPoint - FULLY_OPEN_SNAP_POINT)
+    .maxDuration(100000)
+    .withRef(blockScrollUntilAtTheTopRef);
+
+  const panGesture = Gesture.Pan()
+    .onUpdate(onPanHandlerUpdate)
+    .onEnd(onPanHandlerEnd)
+    .simultaneousWithExternalGesture(nativeViewRef)
+    .withRef(panGestureRef);
+
+  const headerGesture = Gesture.Pan()
+    .onUpdate(onHeaderHandlerUpdate)
+    .onEnd(onHeaderHandlerEnd);
+
+  const bottomSheetAnimatedStyle = useAnimatedStyle(() => {
+    const translateY = bottomSheetTranslateY.value + translationY.value;
+
+    const minTranslateY = Math.max(FULLY_OPEN_SNAP_POINT, translateY);
+    const clampedTranslateY = Math.min(CLOSED_SNAP_POINT, minTranslateY);
+    return {
+      transform: [{ translateY: clampedTranslateY }],
+    };
+  });
+
+  return (
+    <GestureDetector gesture={blockScrollUntilAtTheTop}>
+      <View style={styles.container}>
+        <LoremIpsum words={200} />
+        <Animated.View style={[styles.bottomSheet, bottomSheetAnimatedStyle]}>
+          <GestureDetector gesture={headerGesture}>
+            <View style={styles.header} />
+          </GestureDetector>
+          <GestureDetector
+            gesture={Gesture.Simultaneous(
+              panGesture,
+              blockScrollUntilAtTheTop
+            )}>
```

Comment:
Here the `blockScrollUntilAtTheTop` gesture is attached to both gestures detectors. To just mark a relation without attaching it you can use `simultaneousWithExternalGesture`.